### ISSUE TITLE
Errors in TWW Beta

### ISSUE THREAD

**Issue by yajinni** [nyyr/ArcHUD3#80]
11x ArcHUD3/Core.lua:695: attempt to call global 'UnitBuff' (a nil value)
[string "@ArcHUD3/Core.lua"]:695: in function `TargetAuras'
[string "@ArcHUD3/Core.lua"]:643: in function `?'
[string "@ArcHUD3/Libs/CallbackHandler-1.0-8/CallbackHandler-1.0.lua"]:109: in function <...UD3/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:109>
[string "=[C]"]: ?
[string "@ArcHUD3/Libs/CallbackHandler-1.0-8/CallbackHandler-1.0.lua"]:19: in function <...UD3/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:15>
[string "@ArcHUD3/Libs/CallbackHandler-1.0-8/CallbackHandler-1.0.lua"]:54: in function `Fire'
[string "@ArcHUD3/Libs/AceEvent-3.0-4/AceEvent-3.0.lua"]:120: in function <...ceArcHUD3/Libs/AceEvent-3.0/AceEvent-3.0.lua:119>
[string "=[C]"]: in function `CameraOrSelectOrMoveStop'
[string "CAMERAORSELECTORMOVE"]:4: in function <[string "CAMERAORSELECTORMOVE"]:1>

Locals:
self = <table> {
 defaults = <table> {
 }
 customModuleCount = 0
 modules = <table> {
 }
 defaultModulePrototype = <table> {
 }
 isClassicWoW = false
 hooks = <table> {
 }
 hasHealPrediction = true
 name = "ArcHUD"
 timers = <table> {
 }
 movableFrames = <table> {
 }
 classic = false
 PlayerIsInCombat = false
 baseName = "ArcHUD3"
 defaultModuleState = true
 TargetHUD = ArcHUDFrameTargetHUD {
 }
 Nameplates = <table> {
 }
 isClassicTbc = false
 isClassicCata = false
 RepColor = <table> {
 }
 enabledState = true
 updating = false
 authors = "nyyr, Nenie"
 defaultModuleLibraries = <table> {
 }
 Enabled = true
 customModules = <table> {
 }
 PetIsInCombat = false
 PlayerIsRegenOn = true
 ClassColor = <table> {
 }
 configFrameModules = Frame {
 }
 version = "v7.0.7 (b09044c)"
 configFrameCore = Frame {
 }
 db = <table> {
 }
 configOptionsTableCustomModules = <table> {
 }
 codename = "Pandemic"
 templatePowerRing = <table> {
 }
 configOptionsTableCore = <table> {
 }
 configOptionsTableModules = <table> {
 }
 modulePrototype = <table> {
 }
 hasAbsorbs = true
 isClassicWrath = false
 orderedModules = <table> {
 }
 configOptionsTableCmd = <table> {
 }
}
event = nil
arg1 = "target"
unit = "target"
i = nil
icon = nil
buff = nil
count = nil
buffType = nil
color = nil
duration = nil
expirationTime = nil
filter = ""
(for index) = 1
(for limit) = 40
(for step) = 1
i = 1
(*temporary) = nil
(*temporary) = "target"
(*temporary) = 1
(*temporary) = ""
(*temporary) = "attempt to call global 'UnitBuff' (a nil value)"

**Comment by rookdorf** [comment]
Tried to check this out, but couldn't repro on live. Might be beta only, which is down right now. Will try to check it later if it comes back
Notes for self: I think it may just be, use GetUnitBuff instead?

**Comment by TreizeEU** [comment]
This lua error appear to still occur on retail, after patch 11.0.2 that went live this week.

[ArcHUD lua error.txt](https://github.com/user-attachments/files/16619035/ArcHUD.lua.error.txt)

https://warcraft.wiki.gg/wiki/Patch_11.0.2/API_changes

According to that page "UnitBuff" needs to be replaced with "C_UnitAuras.GetBuffDataByIndex"? I'm not sure where, though.
And "UnitDebuff" -> "C_UnitAuras.GetDebuffDataByIndex". I changed those on lines 695 and 728, seems to work.

**Comment by rookdorf** [comment]
Yep that's correct. I just reproed on live and made the same swap where I hit it (Stagger) and it cleared it up. I'll go through and update them soon, I'm on vacation atm. Thanks for checking!
